### ISSUE TITLE
Indentation breaks list recognition

### ISSUE THREAD

**Issue by mfriedenhagen** [catcombo/jira2markdown#9]
Hello,

* Thanks for the library, works like a charme in general.
* I am using `jira2markdown` in version 0.2.1 in a poetry project.

* if you use the visual editor in JIRA, you often get a blank in front of the list symbol. It is nonetheless correctly visualized in JIRA.

```
❯ python
Python 3.10.5 (main, Jun 23 2022, 17:15:25) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import jira2markdown
>>> jira2markdown.convert('* Blue\n* Green')
'- Blue\n- Green'
>>> jira2markdown.convert(' * Blue\n * Green')
' \\* Blue\n \\* Green'
```

**Comment by catcombo** [comment]
@mfriedenhagen sorry for the very long answer. Does JIRA add a space before all list characters (including nested list characters)? Does JIRA add only one space or more?

**Comment by mfriedenhagen** [comment]
Hi @catcombo, no problem, we are all busy people :-). To reproduce, create a ticket and use the visual editor. When you then start typing `* item1<ENTER>* item2` and then switch to text mode, the markup will look like this:
```
* item1
 * item2 
```
So it is even inconsistent :-( 
The same happens when you just enter `item1<ENTER>item2<ENTER>`, select the two lines and then click the "Bullet List" icon. First line has no leading space, second one has.

**Comment by catcombo** [comment]
@mfriedenhagen I tried to reproduce it on my company's JIRA in-cloud instance and I wasn't successful with that. There is no text mode, only visual mode. I remember that JIRA looked different 2 years ago, probably visual editor is set by default now for all in-cloud instances.

But I was able to switch to the old ticket view and I did some tests with the old editor. I can confirm that space indents before the list items do not affect the list rendering. I prepared a https://github.com/catcombo/jira2markdown/pull/13 Could you please test it?

**Comment by catcombo** [comment]
Changes are released in version `0.3.1`.
